### PR TITLE
Fix printed file path in PrintingTest.scala

### PR DESCRIPTION
**Pull request by michelou**
This fix solves one test case among several tests still failing on Windows (Win10/Java 8).
```
> sbt "testOnly dotty.tools.dotc.PrintingTest"
[info] Loading settings for project dotty-build-build from build.sbt ...
[info] Loading project definition from W:\dotty\project\project
[info] Loading settings for project dotty-build from build.sbt,plugins.sbt ...
[info] Loading project definition from W:\dotty\project
[info] Loading settings for project dotty from build.sbt ...
[info] Resolving key references (25620 settings) ...
[info] Set current project to dotty (in build file:/W:/dotty/)
[info] Passed: Total 0, Failed 0, Errors 0, Passed 0
[info] No tests to run for dotty-interfaces / Test / testOnly
[info] Passed: Total 0, Failed 0, Errors 0, Passed 0
[info] No tests to run for dotty-library / Test / testOnly
[info] Passed: Total 0, Failed 0, Errors 0, Passed 0
[info] No tests to run for tasty-core / Test / testOnly
[warn] Multiple main classes detected.  Run 'show discoveredMainClasses' to see the list
[info] Passed: Total 0, Failed 0, Errors 0, Passed 0
[info] No tests to run for Test / testOnly
[info] Compiling 1 Scala source to W:\dotty\compiler\target\scala-0.22\test-classes ...
[info] Passed: Total 0, Failed 0, Errors 0, Passed 0
[info] No tests to run for dotty-sbt-bridge / Test / testOnly
[info] Test run started
[info] Test dotty.tools.dotc.PrintingTest.printing started
Output from 'tests\printing\i620.scala' did not match check file. Actual output:
result of tests\printing\i620.scala after typer:
package O {
  package O.A {
    class D() extends Object() {
      class C() extends Object() {
        protected[D] def a: Int = 0
        private[D] def b: Int = 0
        private[this] def c: Int = 0
        protected def d: Int = 0
        private[A] def e: Int = 0
        protected[A] def f: Int = 0
        def g: Int = 0
        def g1(t: Int): Int = 0
        def (c: D.this.C) g1: Int = 0
      }
      private[D] class E() extends Object() {}
      private[this] class F() extends Object() {}
      private[A] class G() extends Object() {}
      protected[D] class H() extends Object() {}
      protected class I() extends Object() {}
      protected[A] class J() extends Object() {}
      class K() extends Object() {}
      protected[D] val a: Int = 0
      private[D] val b: Int = 0
      private[this] val c: Int = 0
      protected val d: Int = 0
      private[A] val e: Int = 0
      protected[A] val f: Int = 0
      val g: Int = 0
    }
  }
}


[error] Test dotty.tools.dotc.PrintingTest.printing failed: java.lang.AssertionError: assertion failed: Pass: 0, Failed: 1, took 4.0 sec
Test output dumped in: tests\printing\i620.check.out
  See diff of the checkfile (`brew install icdiff` for colored diff)
    > diff tests\printing\i620.check tests\printing\i620.check.out
  Replace checkfile with current output
    > mv tests\printing\i620.check.out tests\printing\i620.check

[error]     at dotty.DottyPredef$.assertFail(DottyPredef.scala:17)
[error]     at dotty.tools.dotc.PrintingTest.printing(PrintingTest.scala:59)
[error]     ...
[info] Test run finished: 1 failed, 0 ignored, 1 total, 4.391s
[error] Failed: Total 1, Failed 1, Errors 0, Passed 0
[error] Failed tests:
[error]         dotty.tools.dotc.PrintingTest
[info] Passed: Total 0, Failed 0, Errors 0, Passed 0
[info] No tests to run for dotty-doc / Test / testOnly
[error] (dotty-compiler / Test / testOnly) sbt.TestsFailedException: Tests unsuccessful
[error] Total time: 19 s, completed 11 Feb 2020 21:28:11
```
On Windows both files differ only in the first line for test `i620.scala` :
- 1st line in check file: `result of tests/printing/i620.scala after typer:`
- 1st line in output file: `result of tests\printing\i620.scala after typer:`

Replacing selectively the file separator on Windows does solve the issue:
```
> sbt "testOnly dotty.tools.dotc.PrintingTest"
[info] Loading settings for project dotty-i620-build-build from build.sbt ...
[info] Loading project definition from W:\dotty-i620\project\project
[info] Loading settings for project dotty-i620-build from build.sbt,plugins.sbt ...
[info] Loading project definition from W:\dotty-i620\project
[info] Loading settings for project dotty from build.sbt ...
[info] Resolving key references (25620 settings) ...
[info] Set current project to dotty (in build file:/W:/dotty-i620/)
[info] Passed: Total 0, Failed 0, Errors 0, Passed 0
[info] No tests to run for dotty-library / Test / testOnly
[info] Passed: Total 0, Failed 0, Errors 0, Passed 0
[info] No tests to run for dotty-interfaces / Test / testOnly
[info] Passed: Total 0, Failed 0, Errors 0, Passed 0
[info] No tests to run for tasty-core / Test / testOnly
[warn] Multiple main classes detected.  Run 'show discoveredMainClasses' to see the list
[info] Passed: Total 0, Failed 0, Errors 0, Passed 0
[info] No tests to run for Test / testOnly
[info] Passed: Total 0, Failed 0, Errors 0, Passed 0
[info] No tests to run for dotty-sbt-bridge / Test / testOnly
[info] Passed: Total 0, Failed 0, Errors 0, Passed 0
[info] No tests to run for dotty-doc / Test / testOnly
[info] Test run started
[info] Test dotty.tools.dotc.PrintingTest.printing started
Pass: 1, Failed: 0
[info] Test run finished: 0 failed, 0 ignored, 1 total, 4.0s
[info] Passed: Total 1, Failed 0, Errors 0, Passed 1
[success] Total time: 12 s, completed 11 Feb 2020 21:25:30
```

### CODE REVIEW

**Comment by michelou** [comment]
__@liufengyun__: Changes in this request provide a solution limited to the test suite and do not deal with the cause at its roots. Indeed, from the source code in file [`Run.scala`](https://github.com/lampepfl/dotty/blob/master/compiler/src/dotty/tools/dotc/Run.scala#L229) 
```
229: ctx.echo(s"result of $unit after $squashedPhase:")
```
we see that the string representation of variable `unit: CompilationUnit` is computed as follows:
```scala
CompilationUnit -->
     class CompilationUnit protected (val source: SourceFile) {
       override def toString: String = source.toString
       /* ... */
      }
SourceFile -->
     class SourceFile(file: AbstractFile, /* ...*/) extends interfaces.SourceFile {
       override def path: String = file.path
       /* ... */
       override def toString: String = file.toString
       /* ... */
     }
AbstractFile -->
    abstract class AbstractFile extends /* ... */ {
      def path: String
      /* ... */
      override def toString(): String = path
    }
```
<sup>References: [CompilationUnit.scala](https://github.com/lampepfl/dotty/blob/master/compiler/src/dotty/tools/dotc/CompilationUnit.scala), [SourceFile.scala](https://github.com/lampepfl/dotty/blob/master/compiler/src/dotty/tools/dotc/util/SourceFile.scala), [AbstractFile.scala](https://github.com/lampepfl/dotty/blob/master/compiler/src/dotty/tools/io/AbstractFile.scala).</sup>
I don't know for now what would be the best way to modify the result of method `toString`, so please discuss/decide with the team if further investigation is wanted resp. if an idea does emerge on your side.

File: compiler/test/dotty/tools/dotc/printing/PrintingTest.scala [hunk 1]
```diff
@@ -21,18 +21,16 @@ class PrintingTest {
...
   private def compileFile(path: JPath): Boolean = {
     val baseFilePath  = path.toString.stripSuffix(".scala")
     val checkFilePath = baseFilePath + ".check"
     val byteStream    = new ByteArrayOutputStream()
     val reporter = TestReporter.reporter(new PrintStream(byteStream), INFO)
 
+    val jOutFilePath = Paths.get(baseFilePath + ".check.out")
+    if (Files.exists(jOutFilePath))
+      try { Files.delete(jOutFilePath) } catch { case _: Exception => () }
+
```
Comment (liufengyun) [inline compiler/test/dotty/tools/dotc/printing/PrintingTest.scala]: Will it be better to remove the file in `FileDiff.checkAndDump`?

Comment (michelou) [reply compiler/test/dotty/tools/dotc/printing/PrintingTest.scala]: You're right. I can do it in [FileDiff.checkAndDump](https://github.com/lampepfl/dotty/blob/5374d91fc103191455360cdd6a4c69a0c7c50189/compiler/test/dotty/tools/vulpix/FileDiff.scala#L46).

I'll move the code.

**Comment by liufengyun** [comment]
LGTM 👍 

Maybe rebase to remove histories like `Merge remote-tracking branch 'upstream/master' into dotty-i620`.

**Pull request closed without merging**